server/filestore.go

```diff
@@ -4073,6 +4071,14 @@ func (mb *msgBlock) removeSeqPerSubject(subj string, seq uint64) {
 	if seq != ss.First {
 		return
 	}
+
+	// Here what we are removing is the first message.
+	// If we only have one message left we can simply assign it to last.
+	if ss.Msgs == 1 {
+		ss.First = ss.Last
```

Comment:
Just checking: is it that or ss.Last+1? Because if I remember, when we expire/remove all, then First is set to Last+1 to indicate that there are no message left.

Reply:
So we removed the candidate msg above, since we are here we know that the one removed was the first, and we only have 1 msg left so first should equal last.

Reply:
If no msgs left we remove a few lines above the complete entry.

Reply:
Oh right, ss.Msgs == 1 is *after* the remove. My bad.